src/main/java/practice/StreamPractice.java

```diff
@@ -1,80 +1,80 @@
 package practice;
 
-import java.util.Collections;
+import filter.GetWorkablePeopleFilter;
+import filter.SelectMenByAgeFilter;
+import java.util.Arrays;
 import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
 import model.Candidate;
+import model.Cat;
 import model.Person;
 
 public class StreamPractice {
-    /**
-     * Given list of strings where each element contains 1+ numbers:
-     * input = {"5,30,100", "0,22,7", ...}
-     * return min integer value. One more thing - we're interested in even numbers.
-     * If there is no needed data throw RuntimeException with message
-     * "Can't get min value from list: < Here is our input 'numbers' >"
-     */
+    public static final GetWorkablePeopleFilter getWorkablePeopleFilter =
+            new GetWorkablePeopleFilter();
+    public static final SelectMenByAgeFilter selectMenByAgeFilter =
+            new SelectMenByAgeFilter();
+    private static final CandidateValidator candidateValidator = new CandidateValidator();
+
     public int findMinEvenNumber(List<String> numbers) {
-        return 0;
+        return numbers.stream()
+                .flatMap(s -> Arrays.stream(s.split(",")))
+                .mapToInt(Integer::valueOf)
+                .filter(i -> i % 2 == 0)
+                .min()
+                .orElseThrow(() -> new RuntimeException("Can't get min value from list: "
+                        + numbers));
     }
 
-    /**
-     * Given a List of Integer numbers,
-     * return the average of all odd numbers from the list or throw NoSuchElementException.
-     * But before that subtract 1 from each element on an odd position (having the odd index).
-     */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+        List<Integer> collect = IntStream.range(0, numbers.size())
+                .filter(i -> i % 2 != 0)
+                .map(i -> numbers.set(i, numbers.get(i) - 1))
+                .boxed()
+                .collect(Collectors.toList());
+        return numbers.stream()
+                .filter(i -> i % 2 != 0)
+                .mapToDouble(s -> s)
+                .average()
+                .getAsDouble();
```

Comment:
you can simplify this code.  You don't need two filters. 
```suggestion
       .map(i -> i % 2 != 0 ? numbers.get(i) - 1 : numbers.get(i))
                .filter(i -> i % 2 != 0)
```
I think it looks better